src/com/cloudogu/ces/cesbuildlib/SCMManager.groovy

```diff
@@ -0,0 +1,125 @@
+    def pullRequest
+    for (Map pr : getPullRequests()) {
+      if (pr["title"] == title) {
+        pullRequest = pr
+      }
+    }
+
+    return pullRequest
+  }
+
+  Object getPullRequests() {
+    String curlCommand = """curl -w "%{http_code}" ${getCurlAuthParam()} -H 'Content-Type: application/vnd.scmm-pullRequestCollection+json;v=2' https://${this.repositoryUrl}"""
+    String httpResponse = script.sh returnStdout: true, script: curlCommand
+
+    //get only the http code (3 characters) of the response.
+    String httpCode = httpResponse.reverse().take(3).reverse()
+    //drop the http code (last 3 characters) of this string
+    String pullRequests = httpResponse.reverse().drop(3).reverse()
+
+    httpCode = httpCode.trim()
+    script.echo "Getting all pull requests yields httpCode: ${httpCode}"
+    if (httpCode != "200") {
+      script.unstable 'Could not create pull request'
+    }
+
+    def prsAsJson = script.readJSON text: pullRequests
+    return prsAsJson["_embedded"]["pullRequests"]
+  }
+
+  String createPullRequest(String source, String target, String title, String description) {
+    String data = """{"title": "${title}", "description": "${description}", "source": "${source}", "target": "${target}"}"""
+    String curlCommand = """curl -i -X POST ${getCurlAuthParam()} -H 'Content-Type: application/vnd.scmm-pullRequest+json;v=2' -d '${data}' https://${this.repositoryUrl}"""
+    String header = script.sh returnStdout: true, script: curlCommand
+
+    String[] splitHeader = header.split("\n")
+
+    String httpCode = ""
+    for(String line: splitHeader) {
+      if(line.contains("HTTP/2")){
+        String[] httpCodeFull = line.split(" ")
+        httpCode = httpCodeFull[1]
+      }
+    }
+
+    httpCode = httpCode.trim()
+    script.echo "Creating pull request yields httpCode: ${httpCode}"
+    if (httpCode != "201") {
+      script.unstable 'Could not create pull request'
+    }
+
+    String pullRequestId = ""
+    for(String line: splitHeader) {
+      if(line.contains("location:")){
+        String[] httpCodeFull = line.split("/")
+        pullRequestId = httpCodeFull[httpCodeFull.size() - 1]
+        pullRequestId.trim()
+      }
+    }
+
+    return pullRequestId
+  }
+
+  void updateDescription(String pullRequestId, String title, String description) {
+    // In order to update the description put in also the title. Otherwise the title is overwritten with an empty string.
+    String data = """{"title": "${title}","description": "${description}"}"""
+    String curlCommand = """curl -X PUT -w "%{http_code}" ${getCurlAuthParam()} -H 'Content-Type: application/vnd.scmm-pullRequest+json;v=2' -d '${data}' https://${this.repositoryUrl}/${pullRequestId}"""
+    String httpCode = script.sh returnStdout: true, script: curlCommand
+
+    httpCode = httpCode.trim()
+    script.echo "Description update yields http_code: ${httpCode}"
+    if (httpCode != "204") {
+      script.unstable 'Could not update description'
```

Comment:
I think we shouldn't patronize the user here and set the build unstable. The caller should decide what to do in case of failure. So we should return an exception or boolean here instead.

Reply:
See comment [above](https://github.com/cloudogu/ces-build-lib/pull/50#pullrequestreview-575275059)